cli/waiter/subcommands/create.py

```diff
@@ -0,0 +1,131 @@
+import argparse
+import logging
+
+import requests
+
+from waiter import terminal, http
+from waiter.querying import get_token
+from waiter.util import guard_no_cluster, print_info
+
+TRUE_STRINGS = ('yes', 'true')
+FALSE_STRINGS = ('no', 'false')
+BOOL_STRINGS = TRUE_STRINGS + FALSE_STRINGS
+
+create_parser = None
+
+
+def print_post_result(resp):
+    """Prints the result of a token POST"""
+    resp_json = resp.json()
+    if 'message' in resp_json:
+        message = resp_json['message']
+        print_info(f'{message}.')
+        return
+
+    if 'waiter-error' in resp_json and 'message' in resp_json['waiter-error']:
+        message = resp_json['waiter-error']['message']
+    else:
+        message = 'Encountered unexpected error'
+    raise Exception(f'{message}.')
```

Comment:
Should we include the CID in the message? Or is it included in some other debug output?

Reply:
It's included if you run with `--verbose`, e.g.:

```bash
$ waiter --verbose create foo
2019-03-06 13:18:21,495 [DEBUG] [root] args: {'cluster': None, 'url': None, 'config': None, 'action': 'create', 'name': None, 'version': None, 'cmd': None, 'cmd-type': None, 'cpus': None, 'mem': None, 'ports': None, 'token': ['foo']}
2019-03-06 13:18:21,495 [DEBUG] [root] attempting to load json configuration from /home/dpo/source/waiter/cli/.waiter.json
2019-03-06 13:18:21,495 [DEBUG] [root] using configuration: {'http': {'retries': 2, 'connect-timeout': 3.05, 'read-timeout': 20, 'modules': {'session-module': 'requests', 'adapters-module': 'requests.adapters'}}, 'metrics': {'disabled': True, 'timeout': 0.15, 'max-retries': 2, 'host': 'localhost', 'port': 8125, 'line-formats': {'count': '{namespace}.{name}:{value}|c'}}, 'clusters': [{'name': 'dev0', 'url': 'http://127.0.0.1:9091/', 'disabled': False, 'default-for-create': True}, {'name': 'dev1', 'url': 'http://127.0.0.1:9191/', 'disabled': True}]}
2019-03-06 13:18:21,495 [DEBUG] [root] using http timeouts: (3.05, 20)
2019-03-06 13:18:21,495 [DEBUG] [urllib3.util.retry] Converted retries value: 2 -> Retry(total=2, connect=None, read=None, redirect=None, status=None)
2019-03-06 13:18:21,495 [DEBUG] [root] create args: {'token': ['foo']}
2019-03-06 13:18:21,495 [INFO] [root] GET http://127.0.0.1:9091/token with params {'token': ['foo']}
2019-03-06 13:18:21,497 [DEBUG] [urllib3.connectionpool] Starting new HTTP connection (1): 127.0.0.1:9091
2019-03-06 13:18:21,502 [DEBUG] [urllib3.connectionpool] http://127.0.0.1:9091 "GET /token?token=foo HTTP/1.1" 404 None
2019-03-06 13:18:21,502 [INFO] [root] GET response: {"waiter-error":{"service-id":null,"cid":"da359784e7-1744e85cf19b44d2","support-info":[{"label":"Waiter on GitHub","link":{"type":"url","value":"http://github.com/twosigma/waiter"}}],"instance-id":null,"host":"127.0.0.1:9091","details":{"headers":{"etag":null},"status":404,"token":"foo"},"status":404,"uri":"/token","principal":"dpo","query-string":"token=foo","timestamp":"2019-03-06T19:18:21.498Z","request-method":"GET","message":"Couldn't find token foo"}}
2019-03-06 13:18:21,502 [WARNING] [root] Unexpected response code 404 for data request. Response body: {"waiter-error":{"service-id":null,"cid":"da359784e7-1744e85cf19b44d2","support-info":[{"label":"Waiter on GitHub","link":{"type":"url","value":"http://github.com/twosigma/waiter"}}],"instance-id":null,"host":"127.0.0.1:9091","details":{"headers":{"etag":null},"status":404,"token":"foo"},"status":404,"uri":"/token","principal":"dpo","query-string":"token=foo","timestamp":"2019-03-06T19:18:21.498Z","request-method":"GET","message":"Couldn't find token foo"}}
Attempting to post on dev0 cluster...
2019-03-06 13:18:21,502 [INFO] [root] POST http://127.0.0.1:9091/token with body {}
2019-03-06 13:18:21,506 [DEBUG] [urllib3.connectionpool] http://127.0.0.1:9091 "POST /token?token=foo HTTP/1.1" 400 None
2019-03-06 13:18:21,507 [INFO] [root] POST response: {"waiter-error":{"service-id":null,"cid":"da35fc0759-6366d5021f6fc16f","support-info":[{"label":"Waiter on GitHub","link":{"type":"url","value":"http://github.com/twosigma/waiter"}}],"instance-id":null,"host":"127.0.0.1:9091","details":{"status":400},"status":400,"uri":"/token","principal":"dpo","query-string":"token=foo","timestamp":"2019-03-06T19:18:21.505Z","request-method":"POST","message":"No parameters provided for foo"}}
2019-03-06 13:18:21,507 [ERROR] [root] exception when running with ['--verbose', 'create', 'foo']
Traceback (most recent call last):
  File "/home/dpo/source/waiter/cli/waiter/__main__.py", line 17, in main
    result = run(args)
  File "/home/dpo/source/waiter/cli/waiter/cli.py", line 87, in run
    result = actions[action](clusters, args, config_path)
  File "/home/dpo/source/waiter/cli/waiter/subcommands/create.py", line 81, in create
    return create_or_update(cluster, token_name, token_fields)
  File "/home/dpo/source/waiter/cli/waiter/subcommands/create.py", line 48, in create_or_update
    print_post_result(resp)
  File "/home/dpo/source/waiter/cli/waiter/subcommands/create.py", line 29, in print_post_result
    raise Exception(f'{message}.')
Exception: No parameters provided for foo.
No parameters provided for foo.
```